src/redux/slices/arcgis-auth-slice.ts

```diff
@@ -0,0 +1,46 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+import { RootState } from "../store";
+import { getAuthenticatedUser, arcGisRequestLogin, arcGisRequestLogout } from '../../utils/arcgis-auth';
+
+// Define a type for the slice state
+export interface ArcGisAuthState {
+  user: string;
+}
+
+const initialState: ArcGisAuthState = {
+  user: getAuthenticatedUser(),
+}
+
+const arcGisAuthSlice = createSlice({
+  name: "arcGisAuth",
+  initialState,
+  reducers: {
+  },
+  extraReducers: (builder) => {
+    builder.addCase(arcGisLogin.fulfilled, (state, action) => {
+      state.user = action.payload || '';
+    })
+      .addCase(arcGisLogout.fulfilled, (state) => {
+        state.user = '';
+      })
+  },
+});
+
+export const arcGisLogin = createAsyncThunk(
+  'arcGisLogin',
+  async () => {
```

Comment:
Can we send a go-back page and return there instead of /auth page?

Reply:
No, i's impossible. The redirect URI must be registered on ArcGIS, and it must be the same both for begin and complete auth.